docs/dependencies.rst

```diff
@@ -0,0 +1,189 @@
+=====================================
+How Test Dependencies Work In ReFrame
+=====================================
+
+Before going into details on how ReFrame treats test dependencies, it is important to understand how tests are actually treated and executed by the runtime.
+Normally, a ReFrame test will be tried for different programming environments and different partitions within the same ReFrame run.
+These are defined in the test's :func:`__init__` method, but it is not this original object that is being executed by the regression test pipeline.
+The following figure explains in more detail the process:
+
+.. figure:: _static/img/reframe-test-cases.svg
+  :align: center
+  :alt: How ReFrame loads and schedules tests for execution.
+
+When ReFrame loads a test from the disk it unconditionally constructs it executing its :func:`__init__` method.
+The practical implication of this is that your test will be instantiated even if it will not be run on the current system.
```

Comment:
```suggestion
The practical implication of this is that your test will be instantiated even if it is not going to run on the current system.
```

Reply:
I will use "even if it will not run on the current system," instead.